cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -158,25 +215,39 @@ class AzuriteEnv : public ::testing::Environment {
     std::cerr << std::endl;
     return Status::OK();
   }
+};
 
-  const std::string& account_name() const { return account_name_; }
-  const std::string& account_key() const { return account_key_; }
-  const Status status() const { return status_; }
-
+class AzureFlatNSEnv : public AzureEnvImpl<AzureFlatNSEnv> {
  private:
-  std::string account_name_;
-  std::string account_key_;
-  bp::child server_process_;
-  Status status_;
```

Comment:
No more awkward `status_` being set on ctor now that `Make` is used to construct the environments.